neoexchange/astrometrics/sources_subs.py

```diff
@@ -1285,6 +1287,42 @@ def fetch_goldstone_targets(page=None, dbg=False):
                 last_year_seen = year
     return radar_objects
 
+def fetch_goldstone_targets(page=None, calendar_format=False, dbg=False):
+    """Fetches and parses the Goldstone list of radar targets, returning a list
+    of object id's for the current year.
+    Takes either a BeautifulSoup page version of the Goldstone target page (from
+    a call to fetch_goldstone_page() - to allow  standalone testing) or  calls
+    this routine and then parses the resulting page.
+    """
+
+    if type(page) != BeautifulSoup:
+        page = fetch_goldstone_csv(page)
+
+    if page is None:
+        return None
+
+    if type(page) == BeautifulSoup:
+        radar_objects = parse_goldstone_page(page)
+    else:
+        current_year = datetime.utcnow().year
+
+        radar_objects = []
+        for row in page:
+            target = str(row['number'])
+            if is_masked(row['number']) is True:
+                target = row['name']
+            start_date = datetime.strptime(row['start (UT)'], '%m/%d/%Y')
+            if calendar_format is True:
+                end_date = datetime.strptime(row['end (UT)'], '%m/%d/%Y')
+                end_date += timedelta(seconds=86399)
+                target = {'target' : target,
+                          'windows' : [{'start' : start_date.strftime("%Y-%m-%dT%H:%M:%S"),
+                                        'end' : end_date.strftime("%Y-%m-%dT%H:%M:%S")}]
+                             }
+            if start_date.year == current_year:
```

Comment:
I know we don't tend to plan very far ahead, but would it make sense to think about uploading the next year's targets in December rather than waiting until January?

Reply:
Maybe within 2 weeks of the end of December, we allow the next year's targets as valid ?

Reply:
might make some sense, though traditionally, those weeks are great for work.